virttest/qemu_devices/qdevices.py

```diff
@@ -960,3 +960,177 @@ def verify_unplug(self, dev_type, monitor):
         if dev_id_name not in out:
             return True
         return False
+
+
+class CharDevice(QCustomDevice):
+    """
+    Qemu Char Device object, hotplug and hotunplug only support via QMP
+    monitor.
+    """
+    backends = [
+        "null", "socket", "socket", "udp", "msmouse", "vc", "ringbuf", "file",
+        "pipe", "pty", "stdio", "serial", "tty", "parallel", "parport",
+        "spicevmc", "spiceport"
+    ]
+
+    def __init__(self,
+                 params=None,
+                 aobject=None,
+                 parent_bus=None,
+                 child_bus=None):
+        backend = params.get("backend", "socket")
+        self.verify_supported_backend(backend)
+        options = self.get_supported_options(backend)
+        params = params.copy_from_keys(options)
+        params = self.format_params(params)
+        params['backend'] = backend
```

Comment:
no, only support backend can be store in params, see line#982, and line#981 maybe got a default values 'socket'